src/include/ipc/probe.h

```diff
@@ -0,0 +1,106 @@
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+#define PROBE_FMT_TYPE_SHIFT		31
+#define PROBE_STANDARD_TYPE_SHIFT	27
+#define PROBE_AUDIO_FMT_SHIFT		23
+#define PROBE_SAMPLE_RATE_SHIFT		19
+#define PROBE_NB_CHANNELS_SHIFT		14
+#define PROBE_SAMPLE_SIZE_SHIFT		12
+#define PROBE_CONTAINER_SIZE_SHIFT	10
+#define PROBE_SAMPLE_FMT_SHIFT		9
+#define PROBE_SAMPLE_END_SHIFT		8
+#define PROBE_INTERLEAVING_ST_SHIFT	7
+
+#define PROBE_FMT_TYPE_MASK		MASK(31, 31)
+#define PROBE_STANDARD_TYPE_MASK	MASK(30, 27)
+#define PROBE_AUDIO_FMT_MASK		MASK(26, 23)
+#define PROBE_SAMPLE_RATE_MASK		MASK(22, 19)
+#define PROBE_NB_CHANNELS_MASK		MASK(18, 14)
+#define PROBE_SAMPLE_SIZE_MASK		MASK(13, 12)
+#define PROBE_CONTAINER_SIZE_MASK	MASK(11, 10)
+#define PROBE_SAMPLE_FMT_MASK		MASK(9, 9)
+#define PROBE_SAMPLE_END_MASK		MASK(8, 8)
+#define PROBE_INTERLEAVING_ST_MASK	MASK(7, 7)
+
+/* Header for data packets sent via compressed PCM from extraction probes */
+struct probe_data_packet {
+	uint32_t sync_word;
+	uint32_t buffer_id;
+	uint32_t format;
+	uint32_t timestamp_low;
+	uint32_t timestamp_high;
+	uint32_t checksum;
+	uint32_t data_size_bytes;
+	uint32_t data[];
+} __attribute__((packed));
+
+struct probe_dma {
+	uint32_t stream_tag;
+	uint32_t dma_buffer_size;
+} __attribute__((packed));
+
+struct probe_point {
+	uint32_t buffer_id;
+	uint32_t purpose;
+	uint32_t stream_tag;
+} __attribute__((packed));
+
+/* DMA SET for probes - SOF_IPC_PROBE_INIT, SOF_IPC_PROBE_DMA_SET */
+struct sof_ipc_probe_dma_set_params {
+	struct sof_ipc_cmd_hdr hdr;
+	struct probe_dma probe_dma[];
+} __attribute__((packed));
+
+/* Reply to GET functions - SOF_IPC_PROBE_DMA_GET, SOF_IPC_PROBE_POINT_GET */
+struct sof_ipc_probe_get_params {
+	struct sof_ipc_reply rhdr;
+	union {
+		struct probe_dma probe_dma[0];
+		struct probe_point probe_point[0];
+	};
+} __attribute__((packed));
+
+/* DMA detach - SOF_IPC_PROBE_DMA_REMOVE */
+struct sof_ipc_probe_dma_remove_params {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t stream_tag[];
+} __attribute__((packed));
+
+/* Connect probe points - SOF_IPC_PROBE_POINT_SET */
+struct sof_ipc_probe_point_set_params {
+	struct sof_ipc_cmd_hdr hdr;
+	struct probe_point probe_point[];
+} __attribute__((packed));
+
+/* Disconnect probe - SOF_IPC_PROBE_POINT_REMOVE */
+struct sof_ipc_probe_point_remove_params {
+	struct sof_ipc_cmd_hdr hdr;
+	uint32_t buffer_id[];
+} __attribute__((packed));
+
```

Comment:
We really need to doxygen these structures (now that doxygen is used and enabled). I know that this also needs done on other IPC files too.

Reply:
Please verify if documentation provided is sufficent.